src/mesh/MeshPacketQueue.cpp

```diff
@@ -139,8 +147,11 @@ bool MeshPacketQueue::replaceLowerPriorityPacket(meshtastic_MeshPacket *p)
         for (; refPacket->tx_after && it != queue.begin(); refPacket = *--it)
             ;
         if (!refPacket->tx_after && refPacket->priority < p->priority) {
+            LOG_WARN("Dropping non-late packet 0x%08x to make room in the TX queue for higher-priority packet 0x%08x",
+                     refPacket->id, p->id);
             packetPool.release(refPacket);
```

Comment:
There is no call to erase the refPacket from the queue. If the queue was full, this will loop back into replace lower priority packet.

Reply:
Thank you. That explains the loop we were seeing earlier 🙂

Reply:
No worries! It took me a few times to spot it as well. I kept assuming packetPool.release was taking care removing it.

Reply:
Resolved in [11dc10b](https://github.com/meshtastic/firmware/pull/5827/commits/11dc10b1e0219a55d33e9650666f1eeaba31de79).